### PR TITLE
Fees on top

### PR DESCRIPTION
**Pull request by sbinlondon**
Pushing up work as I go along.

Done:

* basic layout based on the current design
* calculates the fee options based on the contribution amount
* calculates the total based on fees + contribution amount
* clear the select value when you choose a new contribution amount
* make the select a controlled component and display the 10% option as the default value
* needs to update the total amount seen in the steps progress box and in the component
* implement what happens when you select 'other'
* hook it up to the backend

Relates opencollective/opencollective#3105

### CODE REVIEW

**Comment by piamancini** [comment]
looks great on the frontend Kate! Only thing, it's missing the phrase under the box about the c3 not sure if you are adding that later.

**Comment by sbinlondon** [comment]
> looks great on the frontend Kate! Only thing, it's missing the phrase under the box about the c3 not sure if you are adding that later.

Yep, still in progress so will add later. I think @znarf had some thoughts on how to implement that specifically.

File: components/contribution-flow/index.js [hunk 1]
```diff
@@ -798,6 +815,22 @@ class CreateOrderPage extends React.Component {
...
+      ['15%', 0.15],
+      ['20%', 0.2],
+    ]);
+    const percentArray = Array.from(map, x => ({
+      label: `${formatCurrency(x[1] * amount, this.getCurrency())} (${x[0]})`,
+      value: x[1] * amount,
+    }));
+    const otherOption = { label: 'I do not want to contribute', value: 0 };
+    const noContributionOption = { label: 'Other', value: 0 };
+    percentArray.push(otherOption, noContributionOption);
```
Comment (kewitz) [inline components/contribution-flow/index.js:830]: 🍨 I wouldn't explicitly declare `otherOption` and `noContributionOption` if I'm using it only once. I would call push with the objects separated on different lines.

```js
percentArray.push(
  { label: 'I do not want to contribute', value: 0 },
  { label: 'Other', value: 0 },
);
```

File: components/contribution-flow/index.js [hunk 2]
```diff
@@ -798,6 +815,22 @@ class CreateOrderPage extends React.Component {
...
+      ['10%', 0.1],
+      ['15%', 0.15],
+      ['20%', 0.2],
+    ]);
+    const percentArray = Array.from(map, x => ({
+      label: `${formatCurrency(x[1] * amount, this.getCurrency())} (${x[0]})`,
+      value: x[1] * amount,
+    }));
+    const otherOption = { label: 'I do not want to contribute', value: 0 };
+    const noContributionOption = { label: 'Other', value: 0 };
```
Comment (kewitz) [inline components/contribution-flow/index.js:829]: Any way we can internationalize these labels?

File: components/contribution-flow/index.js [hunk 3]
```diff
@@ -872,6 +906,50 @@ class CreateOrderPage extends React.Component {
               customData={customData}
               onCustomFieldsChange={this.handleCustomFieldsChange}
             />
+            <FeesOnTopContainer p={3} mt={3}>
+              <Box maxWidth="75%">
+                <P fontSize="Caption" my={2}>
+                  <FormattedMessage
+                    defaultMessage="Open Collective has waived 100% of our fees for COVID-19 Relief Collectives. We rely on the generosity
```
Comment (znarf) [inline components/contribution-flow/index.js:913]: It's already a detail, sorry, but beware of the line break within the string, remove it.

File: components/contribution-flow/index.js (hunk 1 above)
Comment (sbinlondon) [reply components/contribution-flow/index.js:830]: I did think about that option! You're right, I think it makes more sense. Will change.

File: components/contribution-flow/index.js (hunk 2 above)
Comment (sbinlondon) [reply components/contribution-flow/index.js:829]: 100% - good catch

File: components/contribution-flow/index.js (hunk 3 above)
Comment (sbinlondon) [reply components/contribution-flow/index.js:913]: With &nbsp; between the sentences you mean?

Comment (znarf) [reply components/contribution-flow/index.js:917]: Nope, there should be no line break in the `defaultMessage`, will propose the change.

File: components/contribution-flow/index.js [hunk 4]
```diff
@@ -872,6 +910,55 @@ class CreateOrderPage extends React.Component {
...
+                of contributors like you to keep offering this to Collectives."
+                    id="feesOnTop.info"
+                  />
```
Comment (znarf) [inline components/contribution-flow/index.js:920]:
```suggestion
                  <FormattedMessage
                    defaultMessage="Open Collective has waived 100% of our fees for COVID-19 Relief Collectives. We rely on the generosity of contributors like you to keep offering this to Collectives."
                    id="feesOnTop.info"
                  />
```

Comment (sbinlondon) [reply components/contribution-flow/index.js]: Oh, that's a weird formatting. I wonder why it did that 🤔

**Comment by znarf** [comment]
To move forward that project today:

- Handle "Other" input.
- Only display the "Fees On Top" if the Host opted in for the option, maybe `host.settings.feesOnTop = true`. AND if the collective `platformFeePercent = 0`.
- Display the Message about deductible donations if the Host opted in for the option. Maybe `host.settings.taxDeductibleDonations = true`

If  "Fees On Top" is not activated, make sure to NOT send `platformFee` or send it as `null`.

**Comment by sbinlondon** [comment]
> To move forward that project today:
> 
>     * Handle "Other" input.
> 
>     * Only display the "Fees On Top" if the Host opted in for the option, maybe `host.settings.feesOnTop = true`. AND if the collective `platformFeePercent = 0`.
> 
>     * Display the Message about deductible donations if the Host opted in for the option. Maybe `host.settings.taxDeductibleDonations = true`
> 
> 
> If "Fees On Top" is not activated, make sure to NOT send `platformFee` or send it as `null`.

For the `host.settings` stuff is that something you'll be adding on the backend?

**Comment by znarf** [comment]
@sbinlondon no, these settings will be manually added in the DB (for the Foundation host when in production), so you should just edit the host you're testing with and add these settings.

**Comment by znarf** [comment]
Beware, just rebased from master.

**Pull request merged**